web3/utils/transactions.py

```diff
@@ -92,8 +92,28 @@ def get_required_transaction(web3, transaction_hash):
 
 
 def extract_valid_transaction_params(transaction_params):
-    return {key: transaction_params[key]
-            for key in VALID_TRANSACTION_PARAMS if key in transaction_params}
+    extracted_params = {key: transaction_params[key]
+                        for key in VALID_TRANSACTION_PARAMS if key in transaction_params}
+
+    if extracted_params.get('data') is not None:
+        if transaction_params.get('input') is not None:
+            if extracted_params['data'] != transaction_params['input']:
+                msg = 'failure to handle this transaction due to both "input: {}" and'
+                msg += ' "data: {}" are populated. You need to resolve this conflict.'
+                err_vals = (transaction_params['input'], extracted_params['data'])
+                raise AttributeError(msg.format(*err_vals))
+            else:
+                return extracted_params
+        else:
+            return extracted_params
+    elif extracted_params.get('data') is None:
+        if transaction_params.get('input') is not None:
+            extracted_params['data'] = transaction_params['input']
+            return extracted_params
```

Comment:
Just one last piece of mutability that we can eliminate and collapse into one line:

```py
return assoc(extracted_params, 'data', transaction_params['input'])
```

Reply:
@ carver i knew i forgot something ... AttributeError or general Exception ?
```python
exc_msg = 'extract_valid_transaction_params failure for transaction parameters: {}'
Exception({exc_msg".format(transaction_parameters))
```